packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -1263,15 +1264,27 @@ class StackViewport extends Viewport implements IStackViewport {
     bitsAllocated,
     numComps,
     numVoxels,
+    isSigned,
   }): void {
     let pixelArray;
+    const { hasNorm16TextureSupport, preferSizeOverAccuracy } =
+      getConfiguration().rendering;
+    const use16BitDataType = hasNorm16TextureSupport || preferSizeOverAccuracy;
+
     switch (bitsAllocated) {
       case 8:
         pixelArray = new Uint8Array(numVoxels * numComps);
         break;
-
       case 16:
-        pixelArray = new Float32Array(numVoxels * numComps);
+        if (use16BitDataType) {
+          if (isSigned) {
+            pixelArray = new Int16Array(numVoxels * numComps);
+          } else {
+            pixelArray = new Uint16Array(numVoxels * numComps);
+          }
+        } else {
+          pixelArray = new Float32Array(numVoxels * numComps);
+        }
 
```

Comment:
I'm confused, why we don't have the complex logic which involved isSigned and isNegative inside the volume loader here?

Reply:
Revisiting, the `_checkVTKImageDataMatchesCornerstoneImage` should not need to check length and checking type is correct. The isSigned logic is removed in favor of just checking the dtype in the case use16bit is true.